docs/book/programs/tracepoints.md

```diff
@@ -1,3 +1,194 @@
+```
+
+There are 2180 events availble!
+
+In order to avoid name collision, the pattern followed is "\<subsystem name\>:\<tracepoint name\>", as can be seen from the output above.
+
+## Example project
+
+To illustrate tracepoints using Aya, let's write a program which informs us whenever any file is opened by any process running in the system. 
+The kernel tracepoint `syscalls:sys_openat` lets us do this. We will use `cargo generate` utility to build the template code structure:
+
+```bash
+cargo generate https://github.com/aya-rs/aya-template
+ Favorite `https://github.com/aya-rs/aya-template` not found in config, using it as a git repository: https://github.com/aya-rs/aya-template
+ Project Name: aya-tracepoint-echo-open
+ Destination: /home/nirmalya/Workspace-Rust/eBPF/my-second-ebpf/aya-tracepoint-echo-open ...
+ project-name: aya-tracepoint-echo-open ...
+ Generating template ...
+✔  Which type of eBPF program? · tracepoint
+ Which tracepoint category? (e.g sched, net etc...): syscalls
+ Which tracepoint name? (e.g sched_switch, net_dev_queue): sys_enter_openat
+ Moving generated files into: `<Current working directory>/aya-tracepoint-echo-open`...
+ Initializing a fresh Git repository
+ Done! New project created <Current working directory>/aya-tracepoint-echo-open
+```
+
+Note that the project's name is _aya-tracepoint-echo-open_, type of eBPF program is _tracepoint_ (from the menu), category is _syscalls_ and name of tracepoint is _sys_enter_openat_ .
+
+The directory structure is similar to what is described [here](https://aya-rs.dev/book/start/#the-lifecycle-of-an-ebpf-program).
+
+To build the application, move to the directory `aya-tracepoint-echo-open` and fire the following commands:
+
+```bash
+# First, build the application
+cargo xtask build-ebp
+# And, then run
+RUST_LOG=info cargo xtask run
+```
+
+The output on the screen will be:
+
+```bash
+[2024-05-12T02:30:29Z INFO  aya_tracepoint_echo_open] Waiting for Ctrl-C...
+[2024-05-12T02:30:29Z INFO  aya_tracepoint_echo_open] tracepoint sys_enter_openat called
+[2024-05-12T02:30:29Z INFO  aya_tracepoint_echo_open] tracepoint sys_enter_openat called
+[2024-05-12T02:30:29Z INFO  aya_tracepoint_echo_open] tracepoint sys_enter_openat called
+[2024-05-12T02:30:29Z INFO  aya_tracepoint_echo_open] tracepoint sys_enter_openat called
+....
+```
+
+So, the program is running but it is incomplete. We don't know which files are being opened. Let's modify the code, to see the names of those files.
+
+## Design (attempt 1)
+
+For this demo program, we are going to rely on aya-log to print filenames from the eBPF program. 
+
+Because at the tracepoint `sys_enter_openat`, the kernel knows the name of the file (the path to the file, including relative path), we should be able to ask kernel to share that with us. Then, we can print the complete name of the file that is being opened.
+
+For this to happen, such an approach will work:
+
+```rust
+// Source: aya-tracepoint-echo-open-ebpf/src/main.rs
```

Comment:
Please don't include the whole code here. Let's always include the source file, like you did in line 173.

Reply:
Yes, I agree. I will move the whole function into the source code, name it appropriately and leave a link to it in the Markdown file.